resolwe/flow/serializers/collection.py

```diff
@@ -18,6 +20,28 @@ class CollectionSerializer(ResolweBaseSerializer):
         allow_null=True,
         required=False,
     )
+    data_count = serializers.IntegerField()
+    status = serializers.SerializerMethodField()
+
+    def get_status(self, collection):
+        """Return status of the collection based on the status of data objects."""
+        if collection.data_count == 0:
+            return None
+
+        if collection.data_error_count:
+            return Data.STATUS_ERROR
+        if collection.data_uploading_count:
+            return Data.STATUS_UPLOADING
+        if collection.data_processing_count:
+            return Data.STATUS_PROCESSING
+        if collection.data_waiting_count:
+            return Data.STATUS_WAITING
+        if collection.data_resolving_count:
+            return Data.STATUS_RESOLVING
+        if collection.data_done_count == collection.data_count:
+            return Data.STATUS_DONE
+
+        return Data.STATUS_ERROR
```

Comment:
In which case is this returned? If there is no specific case, I would prefer to return `None`.

Reply:
This line of code should never be reached so I'm thinking that raising an exception would be more appropriate.

Reply:
It will be reached if we introduce a new Data status. I think it would be better to return `None` than to raise an error in that case.

You can remove `if collection.data_count == 0:` at the top and return `None` here.

Reply:
If we introduce a new data status, then it would be nice to log if the status cannot be determined.